src/js/webrequest.js

```diff
@@ -528,7 +535,12 @@ function checkAction(tabId, url, quiet, frameId){
   var action = badger.storage.getBestAction(requestHost);
 
   if (action && ! quiet) {
-    badger.logTrackerOnTab(tabId, requestHost, action);
+    let update = badger.logThirdPartyOriginOnTab(tabId, requestHost, action);
+    if (update) {
+      setTimeout(function() {
+        badger.updateCount(tabId);
```

Comment:
Actually I think the `updateCount` function can probably removed completely, and the logic from it can be moved into `logThirdPartyOriginOnTab`, otherwise a lot of the work it does is redundant.